Support Runtime version 3

AndyBitz opened the pull request:
This will allow users to specify the `memory` and `maxDuration` properties for Rust Serverless Functions.

AndyBitz committed:
Support Runtime version 3

AndyBitz committed:
Use constant instead

AndyBitz committed:
Change getDefaultCache

mike-engel commented on src/index.ts:
```diff
@@ -289,7 +295,7 @@ export async function build(opts: BuildOptions) {
 	await runUserScripts(entryPath);
 	const extraFiles = await gatherExtraFiles(config.includeFiles, entryPath);
 
-	if (path.extname(entrypoint) === ".toml") {
```
Can you briefly explain this? I haven't had much time to look over the new `runtime` config, but are they moving away from multiple lambdas?

leo replied:
@mike-engel Indeed! In order to prevent people from using Runtimes to server render their frontend (which drastically decreases performance), Runtimes will – from now on - only be able to execute serverless functions (located in the `api` directory).

As you can see, the check still remains, but it's placed behind a version check. That way, we retain backwards compatibility but make the feature unavailable for when you're using zero config serverless functions.

AndyBitz committed:
Fix tests and add more tests

AndyBitz commented:
I've adjusted the tests and added some more, this is from my local run:

![image](https://user-images.githubusercontent.com/16088670/68634695-9ff01a00-04f6-11ea-951b-7e16e2e12146.png)

leo reviewed (changes requested):
The readme is not adjusted

AndyBitz committed:
Adjust README
